### PR TITLE
Improve acceleration / deceleration , added global parameters

### PR DESCRIPTION
**Pull request by d2macster**
This PR is to address two issues with waypoint_updater.py logic:
1. eliminate hard-coded max allowed speed, and instead read  velocity rosparam from waypoint_loader
2. eliminate hard-coded max_accel and max_decel parameters, and instead read those parameters from twist_controller
3. increase / decrease car velocity proportional to the distance between waypoints, without assuming equal distance between waypoints 

@igolas0 , @TharatchSiri please have a look

### CODE REVIEW

File: ros/src/waypoint_updater/waypoint_updater.py [hunk 1]
```diff
@@ -144,33 +150,38 @@ def prepare_lookahead_waypoints(self):
             end_pos = closest_wp_pos + LOOKAHEAD_WPS - 1  # to build waypoint list with a fixed size
             next_waypoints = list(islice(seq, closest_wp_pos, end_pos))  # list of lookahead waypoints
 
-            max_vel = self.mph_to_mps(50)
             if self.traffic is None or self.traffic == -1:
```
Comment (TharatchSiri) [inline ros/src/waypoint_updater/waypoint_updater.py]: I dont think we need the condition self.traffic is None. I looked into tl_detector.py- either light_wp or self.last_wp does not have None value. So there is no chance that we will come across this condition self.traffic is None. Only self.traffic == -1 is enough here I think.

File: ros/src/waypoint_updater/waypoint_updater.py [hunk 2]
```diff
@@ -144,33 +150,38 @@ def prepare_lookahead_waypoints(self):
...
                 self.is_braking = False
```
Comment (TharatchSiri) [inline ros/src/waypoint_updater/waypoint_updater.py]: What does self.is_braking do? We assign boolean value to self.is_braking but we do nothing with it afterwards in the code. I think we should consider deleting this parameter. The rest of the code will still work fine.

File: ros/src/twist_controller/dbw_node.py [hunk 3]
```diff
@@ -47,6 +47,9 @@ def __init__(self):
         max_lat_accel = rospy.get_param('~max_lat_accel', 3.)
         max_steer_angle = rospy.get_param('~max_steer_angle', 8.)
 
+        rospy.set_param('dbw_decel_limit', decel_limit)
```
Comment (TharatchSiri) [inline ros/src/twist_controller/dbw_node.py]: I'm not sure what you are doing here. Please correct me if I am wrong. So you are creating new parameters 'dbw_decel_limit' and 'dbw_accel_limit' and setting there default values to decel_limit and accel_limit (both obtained from dbw_sim.launch). They share the same values. Why do we have to create new parameter?

File: ros/src/waypoint_updater/waypoint_updater.py [hunk 4]
```diff
@@ -144,33 +150,38 @@ def prepare_lookahead_waypoints(self):
...
-                    target_vel += ACCELERATION
-                    target_vel = min(max_vel, target_vel)
-                    target_vel = max(0.0, target_vel)
-                    self.set_waypoint_velocity(next_waypoints, i, target_vel)
+                    if i == 0:
+                        dist = self.distance(self.current_pose, next_waypoints[0])
+                    else:
+                        dist = self.distance(next_waypoints[i-1].pose.pose, next_waypoints[i])
+
+                    vel += self.accel_limit * dist
```
Comment (TharatchSiri) [inline ros/src/waypoint_updater/waypoint_updater.py]: We introduced new parameter to accelerate velocity but still in the launch file (dbw_sim.launch) where we initialize the value, we set this parameter to 1.0. The car will accelerate at the same speed as the multiplier here is 1. Should we increase it a bit?

File: ros/src/twist_controller/dbw_node.py (hunk 3 above)
Comment (d2macster) [reply ros/src/twist_controller/dbw_node.py]: I didn't know how to access those parameters from Dow_node. I just found, using `rosparam list`  that you can access those via `/dbw_node/...`. fixing the issue

**Pull request merged**